src/engine/loopingcontrol.cpp

```diff
@@ -742,7 +744,6 @@ void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint) {
                     //
                     // If we press 1/2 beatloop we want loop from 50% to 100%,
                     // If I press 1/4 beatloop, we want loop from 50% to 75% etc
-                    double nextBeat = floor(m_pBeats->findNextBeat(cur_pos));
```

Comment:
that was a bug.  the new API guarantees that the beats won't be the same so this should be ok now.